src/overlays/actors/ovl_En_Sth/z_en_sth.h

```diff
@@ -2,34 +2,53 @@
 #define Z_EN_STH_H
 
 #include "global.h"
+#include "objects/object_sth/object_sth.h"
+#include "objects/object_ahg/object_ahg.h" 
+#include "objects/object_mask_truth/object_mask_truth.h"
 
 struct EnSth;
 
 typedef void (*EnSthActionFunc)(struct EnSth*, PlayState*);
 
-#define ENSTH_GET_F(thisx) ((thisx)->params & 0xF)
-#define ENSTH_GET_100(thisx) ((thisx)->params & 0x100)
-
-enum {
-    /* 1 */ ENSTH_F_1 = 1,
-    /* 2 */ ENSTH_F_2,
-    /* 3 */ ENSTH_F_3,
-    /* 4 */ ENSTH_F_4,
-    /* 5 */ ENSTH_F_5,
-};
 
 typedef struct EnSth {
     /* 0x000 */ Actor actor;
     /* 0x144 */ ColliderCylinder collider;
     /* 0x190 */ SkelAnime skelAnime;
-    /* 0x1D4 */ Vec3s jointTable[16];
-    /* 0x234 */ Vec3s morphTable[16];
-    /* 0x294 */ Vec3s unk_294;
-    /* 0x29A */ s16 unk_29A;
-    /* 0x29C */ u16 unk_29C;
-    /* 0x29E */ u8 unk_29E;
-    /* 0x29F */ u8 unk_29F;
+    /* 0x1D4 */ Vec3s jointTable[STH_LIMB_MAX];
+    /* 0x234 */ Vec3s morphTable[STH_LIMB_MAX];
+    /* 0x294 */ Vec3s headRot;
+    /* 0x29A */ s16 curAnimIndex;
+    /* 0x29C */ u16 sthFlags;
+    /* 0x29E */ u8 mainObjectId;
+    /* 0x29F */ u8 maskOfTruthObjectId;
     /* 0x2A0 */ EnSthActionFunc actionFunc;
 } EnSth; // size = 0x2A4
 
+// Note: Vanilla types usually have 0xFEXX typing, but this upper section is unused by the code, reason unknown
+#define STH_GET_TYPE(thisx) ((thisx)->params & 0xF)
+#define STH_GET_SWAMP_BODY(thisx) ((thisx)->params & 0x100)
+
+// The get item for the reward for ocean spiderhouse (wallet, or rupees) is set here
+#define STH_GI_ID(thisx) ((thisx)->home.rot.z)
+
+typedef enum {
+    /* 1 */ STH_TYPE_UNUSED_1 = 1,
+    /* 2 */ STH_TYPE_SWAMP_SPIDER_HOUSE_CURED, // cursed is EnSsh
+    /* 3 */ STH_TYPE_MOON_LOOKING, // South Clock Town, looking up at the moon
+    /* 4 */ STH_TYPE_OCEANSIDE_SPIDER_HOUSE_GREET, // looking for shelter
+    /* 5 */ STH_TYPE_OCEANSIDE_SPIDER_HOUSE_PANIC, // shelter was not enough
+    // Other values: Actor will spawn and animate with arm waving, no further interaction.
+} EnSthTypes;
+
+// This actor has its own flags system
+#define STH_FLAG_CLEAR                          (0)
+#define STH_FLAG_DRAW_TRUTH_MASK                (1 << 0)
+#define STH_FLAG_OCEANSIDE_SPIDER_HOUSE_GREET   (1 << 1)
+#define STH_FLAG_SWAMP_SPIDER_HOUSE_SAVED       (1 << 2) // set, but not read 
+#define STH_FLAG_DISABLE_HEAD_TRACK             (1 << 3)
+
+#define STH_SWAMP_SPIDER_TOKENS_REQUIRED     30
+#define STH_OCEANSIDE_SPIDER_TOKENS_REQUIRED 30
```

Comment:
`en_si.c` was just merged a few days ago: thoughts on which define to use for `Inventory_GetSkullTokenCount` in this spot: seems like this 30 applies for both defines
https://github.com/zeldaret/mm/blob/2cce8f67671c7162a946dd11604f83c9630fbc28/src/overlays/actors/ovl_En_Si/z_en_si.c#L106-L113

Reply:
Damn that suggests they are the same one value, they are treated slightly differently by the code so I thought I could keep them separate, but if the token only checks one value its clearly not.